cpp/src/arrow/python/python-test.cc

```diff
@@ -246,6 +246,16 @@ TEST_F(DecimalTest, FromPythonDecimalRescaleTruncateable) {
   ASSERT_EQ(100, value.low_bits());
 }
 
+TEST_F(DecimalTest, FromPythonNegativeDecimalRescale) {
+  Decimal128 value;
+  OwnedRef python_decimal(this->CreatePythonDecimal("-1.000"));
+  auto type = ::arrow::decimal(10, 9);
+  const auto& decimal_type = static_cast<const DecimalType&>(*type);
+  ASSERT_OK(
+      internal::DecimalFromPythonDecimal(python_decimal.obj(), decimal_type, &value));
+  ASSERT_EQ(-1000000000, (int64_t)(value));
```

Comment:
Don't cast here. The `operator==` definition + implicit casting rules of C++ handle that.